sys/net/application_layer/cord/lc/cord_lc.c

```diff
@@ -191,70 +209,241 @@ static void _on_lookup(unsigned req_state, coap_pkt_t *pdu,
     }
 
-    /* build a CoAP POST request to the RD's lookup interface */
-    res = gcoap_req_init(pkt, buf, sizeof(buf), COAP_METHOD_GET, _rd_lookif);
+    res = gcoap_req_init(&pkt, buf, sizeof(buf), COAP_METHOD_GET, lookif);
     if (res < 0) {
         retval = CORD_LC_ERR;
         goto end;
     }
 
     /* set some packet options */
-    coap_hdr_set_type(pkt->hdr, COAP_TYPE_CON);
+    coap_hdr_set_type(pkt.hdr, COAP_TYPE_CON);
+
+    /* save pointer to result */
+    _result_buf = result;
+    _result_buf_len = maxlen;
+
+    /* specify requested content format */
+    char ct[3];
+    snprintf(ct, sizeof(ct), "%d", content_format);
+    gcoap_add_qstring(&pkt, "ct", ct);
+
+    pkt_len = gcoap_finish(&pkt, 0, COAP_FORMAT_NONE);
+    if (pkt_len < 0) {
+        retval = CORD_LC_ERR;
+        goto end;
+    }
+
+    res = gcoap_req_send2(buf, pkt_len, rd->remote, _on_lookup);
+    if (res < 0) {
+        retval = CORD_LC_ERR;
+        goto end;
+    }
 
+    retval = _sync();
 end:
+    if (retval == CORD_LC_OK) {
+        return strlen(_result_buf);
+    }
+    else {
+        return retval;
+    }
+}
+
+ssize_t cord_lc_res_raw(const cord_lc_rd_t *rd, uint8_t content_format,
+                        char *result, size_t maxlen)
+{
+    _lock();
+    ssize_t retval = _lookup_raw(rd, content_format, result, maxlen, CORD_LC_RES);
     mutex_unlock(&_mutex);
     return retval;
 }
 
-int cord_lc_lookup_send(coap_pkt_t *pkt, sock_udp_ep_t *remote,
-                        char *result, size_t maxlen)
+ssize_t cord_lc_ep_raw(const cord_lc_rd_t *rd, uint8_t content_format,
+                       char *result, size_t maxlen)
 {
-    assert(remote);
+    _lock();
+    ssize_t retval = _lookup_raw(rd, content_format, result, maxlen, CORD_LC_EP);
+    mutex_unlock(&_mutex);
+    return retval;
+}
 
-    int res;
-    int retval;
-    ssize_t pkt_len;
+static void _res_write_attr_val(cord_lc_res_t *res, const char *attr,
+                           const char *val)
+{
+    if (strcmp(attr, "path") == 0) {
+        strncpy(res->path, val, sizeof(res->path));
+    }
+    else if (strcmp(attr, "rt") == 0) {
+        strncpy(res->rt, val, sizeof(res->rt));
+    }
+    else if (strcmp(attr, "if") == 0) {
+        strncpy(res->ifdesc, val, sizeof(res->ifdesc));
+    }
+    else if (strcmp(attr, "anchor") == 0) {
+        strncpy(res->anchor, val, sizeof(res->anchor));
+    }
+
+    /* ignore unrecognized attributes */
+}
+
+static void _ep_write_attr_val(cord_lc_ep_t *ep, const char *attr,
+                           const char *val)
+{
+    if (strcmp(attr, "path") == 0) {
+        strncpy(ep->path, val, sizeof(ep->path));
+    }
+    else if (strcmp(attr, "base") == 0) {
+        strncpy(ep->base, val, sizeof(ep->base));
+    }
+    else if (strcmp(attr, "ep") == 0) {
+        strncpy(ep->ep, val, sizeof(ep->ep));
+    }
+
+    /* ignore unrecognized attributes */
+}
+
+static size_t _parse_res(const char *source, size_t len, cord_lc_res_t *results,
```

Comment:
_parse_res and _parse_ep basically do the same thing - parses the result buffer from _lookup_raw and saves the endpoints/resources to respective structs. I tried to combine the two functions together so that there is less redundant code but cannot find a good solution. Suggestions are welcomed.